Added cucumberjs support and brand new unit tests

nikulkarni opened the pull request.

nikulkarni commented:
@grawk Can you take a look at this?

grawk commented:
Looking now

grawk commented on app/index.js:
```diff
@@ -200,29 +205,58 @@ var NemoGenerator = yeoman.generators.Base.extend({
     //report dir
     this.mkdir(reportDir);
     this.copy('test/functional/report/README.md', reportDir + '/README.md');
-    //spec dir
```
Why did you remove the yahoo example?

grawk commented on app/index.js:
```diff
@@ -149,37 +155,33 @@ var NemoGenerator = yeoman.generators.Base.extend({
-        done();
-        return;
+    //let's update Gruntfile.js if possible
+    var api = require('gruntfile-api'),
```
Please move require to top of file and give more descriptive variable name like gruntfileAPI. Will help to clarify what's going on for folks coming to this fresh.

grawk commented on app/index.js:
```diff
@@ -200,29 +205,58 @@ var NemoGenerator = yeoman.generators.Base.extend({
       return newval;
     };
-    this.template('_loopmocha.js', taskDir + 'loopmocha.js');
+    if(this.testFramework==='mocha'){
```
Missing spaces around ===. I see this in a few places. Obviously stylistic only but pretty standard. If we were using jshint as a part of development, it would complain.. I'll file an issue that jshint should be a part of the build process going forward.

grawk commented:
During unit tests I see some EACCES errors like this:

```bash
npm ERR! Darwin 12.5.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "--save-dev" "cucumber@^0.4.4" "nemo@^v0.2.0" "nemo-view@^v0.2.0" "nemo-drivex@^v0.1.0" "nemo-locatex@^v0.1.0" "nconf@~v0.6.7" "grunt-cucumberjs@^v0.4.1" "grunt-config-dir@^0.3.2"
npm ERR! node v0.10.29
npm ERR! npm  v2.1.17
npm ERR! path ../cucumber/bin/cucumber.js
npm ERR! code EACCES
npm ERR! errno 3

npm ERR! Error: EACCES, symlink '../cucumber/bin/cucumber.js'
npm ERR!  { [Error: EACCES, symlink '../cucumber/bin/cucumber.js'] errno: 3, code: 'EACCES', path: '../cucumber/bin/cucumber.js' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/medelman/src/_nemo/o/generator-nemo/temp/npm-debug.log
```

All tests still pass, but wondering if you see the same.

Also I've added a few comments to move stuff around or add some whitespace. If you can resolve those, and also squash all of this into a single commit, I'll merge then publish a new version.

Thanks for doing this work and I'm sincerely apologetic that I did not look at it sooner!

nikulkarni commented:
@grawk Thanks for reviewing the PR. I understand your other commitments, I just wanted to post a friendly reminder earlier this week so this PR won't fall of your radar :)

I fixed formatting and moved the require to the top. As for the error you are seeing, I've seen it in the past  and it went away when I gave myself full access to .npm directory with this command.
```
sudo chown -R `whoami` ~/.npm
```

grawk commented:
Published as v0.3.0
